main.go

```diff
@@ -71,12 +75,11 @@ func main() {
 	}
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme:             scheme,
-		MetricsBindAddress: metricsAddr,
-		Port:               9443,
-		LeaderElection:     enableLeaderElection,
-		LeaderElectionID:   "91c624a5.openshift.io",
-		Namespace:          watchNamespace,
+		Scheme:           scheme,
+		Metrics:          metricsserver.Options{BindAddress: metricsAddr},
+		LeaderElection:   enableLeaderElection,
+		LeaderElectionID: "91c624a5.openshift.io",
+		Cache:            cache.Options{DefaultNamespaces: map[string]cache.Config{watchNamespace: {}}},
```

Comment:
Just read about this and seems like caching on the ns could improve memory efficiency. Nice work  🎉